lib/codegen.ml

```diff
@@ -0,0 +1,309 @@
+open! Core_kernel
+
+(* open Efsm *)
+
+open! Ppxlib_ast
+open Parsetree
```

Comment:
Is this use of the parsetree stable enough? Does it work with more than one compiler version?

Reply:
This is the ast for ppxlib and is stable